app/assets/javascripts/DropDownMenu/MultiSelectDropDown.js

```diff
@@ -0,0 +1,116 @@
+import React from "react";
+import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
+
+export class MultiSelectDropdown extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {expanded: false, tags: []};
+    this.dropdownRef = React.createRef();
+  }
+
+  componentDidMount() {
+    document.addEventListener("mousedown", this.handleClickOutside);
```

Comment:
Don't add global event listeners. Instead you can try using the `onBlur` React synthetic event.